configure.ac

```diff
@@ -1007,21 +1007,22 @@ AC_SUBST([GETOPT_PATH])
 dnl ===============================================
 dnl Libraries
 dnl ===============================================
-AC_CHECK_LIB(socket, socket)                    dnl -lsocket
-AC_CHECK_LIB(c, dlopen)                         dnl if dlopen is in libc...
-AC_CHECK_LIB(dl, dlopen)                        dnl -ldl (for Linux)
-AC_CHECK_LIB(rt, sched_getscheduler)            dnl -lrt (for Tru64)
-AC_CHECK_LIB(gnugetopt, getopt_long)            dnl -lgnugetopt ( if available )
-AC_CHECK_LIB(pam, pam_start)                    dnl -lpam (if available)
+AC_SEARCH_LIBS([socket], [socket])              dnl -lsocket
+AC_SEARCH_LIBS([sched_getscheduler], [rt],      dnl -lrt (for some old platforms)
+               [test "$ac_cv_search_sched_getscheduler" = "none required" || PC_LIBS_RT="$ac_cv_search_sched_getscheduler"])
+AC_SEARCH_LIBS([dlopen], [dl])                  dnl -ldl (for Linux)
+
+save_LIBS=$LIBS
+AC_SEARCH_LIBS([pam_start], [pam],              dnl -lpam (if available)
+               [test "$ac_cv_search_pam_start" = "none required" || PAMLIBS="$LIBS"])
+AC_SUBST(PAMLIBS)
+LIBS=$save_LIBS
 
 PKG_CHECK_MODULES([UUID], [uuid],
                   [CPPFLAGS="${CPPFLAGS} ${UUID_CFLAGS}"
                    LIBS="${LIBS} ${UUID_LIBS}"])
 
-AC_CHECK_FUNCS([sched_setscheduler])
-AS_IF([test x"$ac_cv_func_sched_setscheduler" != x"yes"],
-      [PC_LIBS_RT=""],
-      [PC_LIBS_RT="-lrt"])
+REQUIRE_FUNC([sched_setscheduler])
```

Comment:
Just noticed we can ditch this altogether by just putting `AC_MSG_FAILURE` in the not-found part